From 51357dc19efbf30328ca05655fbf69886f6e9113 Mon Sep 17 00:00:00 2001
From: Julien Isorce <julien.isorce@chromium.org>
Date: Tue, 05 Dec 2017 23:39:45 +0000
Subject: [PATCH] Allow RGBX for VaapiTFPPicture

Fixes regression on non-ozone platforms and introduced by
  73d609f366ba6a86324048bbad81527f76d237b5
  https://chromium-review.googlesource.com/787290

Note that the format is only used for sanity check. All the logic is
done automatically in the vaapi driver's implementation of vaPutSurface.

Bug: 785201
Cq-Include-Trybots: master.tryserver.chromium.android:android_optional_gpu_tests_rel;master.tryserver.chromium.linux:linux_optional_gpu_tests_rel;master.tryserver.chromium.mac:mac_optional_gpu_tests_rel;master.tryserver.chromium.win:win_optional_gpu_tests_rel
Change-Id: Idc0bdf184874bf2c238e28da6f42f4e3572e9743
Reviewed-on: https://chromium-review.googlesource.com/807928
Reviewed-by: Dan Sanders <sandersd@chromium.org>
Commit-Queue: Julien Isorce <julien.isorce@chromium.org>
Cr-Commit-Position: refs/heads/master@{#521897}
---

diff --git a/media/gpu/vaapi/vaapi_tfp_picture.cc b/media/gpu/vaapi/vaapi_tfp_picture.cc
index e9eecce..bd7823c 100644
--- a/media/gpu/vaapi/vaapi_tfp_picture.cc
+++ b/media/gpu/vaapi/vaapi_tfp_picture.cc
@@ -72,7 +72,8 @@
 bool VaapiTFPPicture::Allocate(gfx::BufferFormat format) {
   DCHECK_CALLED_ON_VALID_SEQUENCE(sequence_checker_);
   if (format != gfx::BufferFormat::BGRX_8888 &&
-      format != gfx::BufferFormat::BGRA_8888) {
+      format != gfx::BufferFormat::BGRA_8888 &&
+      format != gfx::BufferFormat::RGBX_8888) {
     DLOG(ERROR) << "Unsupported format";
     return false;
   }
